pkg/cli/webhook/gitlab.go

```diff
@@ -0,0 +1,116 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/cli/prompt"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/provider"
+	"github.com/xanzy/go-gitlab"
+)
+
+type gitLabConfig struct {
+	Client              *gitlab.Client
+	controllerURL       string
+	projectID           string
+	webhookSecret       string
+	personalAccessToken string
+	APIURL              string
+}
+
+func (gl *gitLabConfig) GetName() string {
+	return provider.ProviderGitLabWebhook
+}
+
+func (gl *gitLabConfig) Run(_ context.Context, opts *Options) (*response, error) {
+	err := gl.askGLWebhookConfig(opts.ControllerURL)
+	if err != nil {
+		return nil, err
+	}
+	gl.APIURL = opts.ProviderAPIURL
+
+	return &response{
+		ControllerURL:       gl.controllerURL,
+		PersonalAccessToken: gl.personalAccessToken,
+		WebhookSecret:       gl.webhookSecret,
+	}, gl.create()
+}
+
+func (gl *gitLabConfig) askGLWebhookConfig(controllerURL string) error {
+	msg := "Please enter project ID for the repository you want to be configured :"
+	if err := prompt.SurveyAskOne(&survey.Input{Message: msg}, &gl.projectID,
+		survey.WithValidator(survey.Required)); err != nil {
+		return err
+	}
+
+	if controllerURL == "" {
+		if err := prompt.SurveyAskOne(&survey.Input{
+			Message: "Please enter your controller public route URL: ",
+		}, &gl.controllerURL, survey.WithValidator(survey.Required)); err != nil {
+			return err
+		}
+	}
+
+	if err := prompt.SurveyAskOne(&survey.Password{
+		Message: "Please enter the secret to configure the webhook for payload validation: ",
+	}, &gl.webhookSecret, survey.WithValidator(survey.Required)); err != nil {
+		return err
+	}
+
+	// nolint:forbidigo
+	fmt.Println("ℹ ️You now need to create a GitLab personal access token with `api` scope")
+	// nolint:forbidigo
+	fmt.Println("ℹ ️Go to this URL to generate one https://gitlab.com/-/profile/personal_access_tokens, see https://is.gd/WIXECN for documentation ")
+	if err := prompt.SurveyAskOne(&survey.Password{
```

Comment:
I wonder if we should try to open the web browser there (the same way we create-app)...

Reply:
so we just open that link in a browser and user will create and then enter on the terminal?

Reply:
ah i don't know maybe it's not that good idea, let's hold on on this